### PR TITLE
Cats Effect 3

### PR DESCRIPTION
**Pull request by tomverran**
This PR upgrades all the modules + dependencies to versions that depend on Cats Effect 3, which has a different type class hierarchy to 2.x. As such this'll be published as a new major version (5.0.0).

### CODE REVIEW

File: docs/docs/docs/metrics.md [hunk 1]
```diff
@@ -40,7 +39,7 @@ object MetricApp extends IOApp {
       metricPrefix = Some("my_app"),
 
       // the address your Datadog agent is listening on
-      agentHost = new InetSocketAddress("localhost", 8125),
+      agentHost = SocketAddress(ip"127.0.0.1", port"8080"),
```
Comment (ericaovo) [inline docs/docs/docs/metrics.md:42]: how come the port has changed?

File: natchez-extras-doobie/src/main/scala/com/ovoenergy/natchez/extras/doobie/TracedTransactor.scala [hunk 2]
```diff
@@ -16,23 +16,22 @@ object TracedTransactor {
...
       .copy(
-        interpret0 = (
-          new KleisliInterpreter[Traced[F, *]] {
+        interpret0 = new KleisliInterpreter[Traced[F, *]] {
 
-            val blocker: Blocker = blocking
-            val trace: Trace.KleisliTrace[F] = Trace[Traced[F, *]]
-            val contextShiftM: ContextShift[Traced[F, *]] = ContextShift.deriveKleisli[F, Span[F]]
-            implicit val asyncM: Async[Traced[F, *]] = Async.catsKleisliAsync[F, Span[F]]
+          implicit val asyncM: WeakAsync[Traced[F, *]] =
```
Comment (ericaovo) [inline natchez-extras-doobie/src/main/scala/com/ovoenergy/natchez/extras/doobie/TracedTransactor.scala:28]: 😱

File: docs/docs/docs/metrics.md (hunk 1 above)
Comment (tomverran) [reply docs/docs/docs/metrics.md:42]: oh yes I should put that back I think :thinking:

File: natchez-extras-fs2/src/test/scala/com/ovoenergy/natchez/extras/fs2/AllocatedSpanTest.scala [hunk 3]
```diff
@@ -62,7 +54,7 @@ class AllocatedSpanTest extends AnyWordSpec with Matchers {
               .covary[IO]
               .through(AllocatedSpan.create(maxOpen = 10)(_ => ep.root("root")))
               .evalTap(_.span.span("foo").use(_ => IO.unit))
-              .metered(1.millisecond)
+              .metered(5.millisecond)
```
Comment (ericaovo) [inline natchez-extras-fs2/src/test/scala/com/ovoenergy/natchez/extras/fs2/AllocatedSpanTest.scala:57]: are you sure?

**Review by ericaovo — APPROVED** [review]

File: natchez-extras-fs2/src/test/scala/com/ovoenergy/natchez/extras/fs2/AllocatedSpanTest.scala (hunk 3 above)
Comment (tomverran) [reply natchez-extras-fs2/src/test/scala/com/ovoenergy/natchez/extras/fs2/AllocatedSpanTest.scala:57]: yes I think there's just some flakiness around things finishing absolutely immediately in these test cases hence this

**Pull request merged**